src/main/Main.hs

```diff
@@ -848,6 +852,16 @@ execCmd ExecOpts {..} go@GlobalOpts{..} = do
                munlockFile lk -- Unlock before transferring control away.
                exec eoEnvSettings cmd args
 
+-- | Evaluate some haskell code inline.
+evalCmd :: ExecOpts -> GlobalOpts -> IO ()
+evalCmd e@ExecOpts { eoArgs = args } go@GlobalOpts {..} =
+    if length args /= 1
+    then error "Expecting ONE string argument to 'eval'"
```

Comment:
Yup, I forgot to mention that I'm not very experienced with `optparse-applicative`. Thinking about it, there should be a way to parse and declare the parse failed if it doesn't match this condition.

Do you think such a solution is viable + possible + worth the trouble?